packages/legend-extension-dsl-data-space/src/stores/query/DataSpaceTemplateQueryPromotionReviewerStore.ts

```diff
@@ -0,0 +1,666 @@
+    } finally {
+      this.loadQueryState.complete();
+    }
+  }
+
+  *loadProject(project: Project): GeneratorFn<void> {
+    this.currentProject = project;
+    this.currentProjectConfigurationStatus = undefined;
+    this.loadWorkspacesState.inProgress();
+    try {
+      this.currentProjectConfigurationStatus =
+        (yield fetchProjectConfigurationStatus(
+          project.projectId,
+          undefined,
+          this.applicationStore,
+          this.sdlcServerClient,
+        )) as ProjectConfigurationStatus;
+
+      const workspacesInConflictResolutionIds = (
+        (yield this.sdlcServerClient.getWorkspacesInConflictResolutionMode(
+          project.projectId,
+          undefined,
+        )) as Workspace[]
+      ).map((workspace) => workspace.workspaceId);
+      this.groupWorkspaces = (
+        (yield this.sdlcServerClient.getGroupWorkspaces(
+          project.projectId,
+        )) as PlainObject<Workspace>[]
+      )
+        .map((v) => Workspace.serialization.fromJson(v))
+        .filter(
+          (workspace) =>
+            // NOTE we don't handle workspaces that only have conflict resolution but no standard workspace
+            // since that indicates bad state of the SDLC server
+            !workspacesInConflictResolutionIds.includes(
+              workspace.workspaceId,
+            ) && workspace.workspaceType === WorkspaceType.GROUP,
+        );
+      this.loadWorkspacesState.pass();
+    } catch (error) {
+      assertErrorThrown(error);
+      this.applicationStore.logService.error(
+        LogEvent.create(LEGEND_STUDIO_APP_EVENT.SDLC_MANAGER_FAILURE),
+        error,
+      );
+      this.applicationStore.notificationService.notifyError(error);
+      this.loadWorkspacesState.fail();
+    }
+  }
+
+  *promoteAsTemplateQuery(): GeneratorFn<void> {
+    const query = this.currentQuery;
+    const project = this.currentProject;
+    if (
+      this.promoteState.isInProgress ||
+      !query ||
+      !this.currentQueryInfo ||
+      !this.currentProjectConfiguration ||
+      !project ||
+      !this.workspaceName ||
+      !this.templateQueryTitle ||
+      !this.dataSpace ||
+      !this.isWorkspaceNameValid ||
+      !this.isTemplateQueryIdValid
+    ) {
+      return;
+    }
+
+    try {
+      this.promoteState.inProgress();
+      // 1. prepare project entities
+      this.applicationStore.alertService.setBlockingAlert({
+        message: `Fetching and updating project...`,
+        prompt: 'Please do not close the application',
+        showLoading: true,
+      });
+      const dataSpaceExecutableTemplate = new DataSpaceExecutableTemplate();
+      dataSpaceExecutableTemplate.id = this.templateQueryId;
+      dataSpaceExecutableTemplate.title = this.templateQueryTitle;
+      dataSpaceExecutableTemplate.description = this.templateQueryDescription;
+      dataSpaceExecutableTemplate.executionContextKey = guaranteeNonNullable(
```

Comment:
lets add a TODO here, we need to update this code once we properly store existing dataspace quieres, they should not have mapping/runtime but executionKeys.